spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,155 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+### Threshold of Reviews on Pull Requests
+
+GitHub allows repository owners to create protection rules that require a
+threshold of approvals before a pull request can be merged. This predicate can
+be used to express meeting such a policy. In this case, GitHub would issue a
+signed attestation identifying all the reviewers for a pull request. The
+subject of the attestation is the underlying commit at the tip of the feature
+branch. This commit, therefore, is one of the parents of the merge commit if it
+is created. If an option that requires rebasing (squash and merge / rebase and
+merge) is selected, the newly created tip commit is used as the attestation
+subject.
+
+## Model
+
+Typically, software development processes incorporate some type of code review.
+Other members of the development team are expected to study the changes
+proposed and approve them, confirming they meet various expectations pertaining
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+As this attestation is generated by a version control service or code review
+service, the signatures are from the service rather than the specific users
+issuing the review.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "name": "<REPOSITORY>",
+            "digest": { "gitCommit": "<HASH>" },
+            "uri": "<URI OF REPOSITORY>"
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewers": ["<ResourceDescriptor>", ...],
+        "targetTip": "<ResourceDescriptor>",
+        "mergeBase": "<ResourceDescriptor>",
+        "reviewLink": "<URI OF REVIEW>",
+        "reviewTime": "<TIMESTAMP>",
+        "annotations": {...}
```

Comment:
Perhaps this falls under the annotations (or could better fit into the CREV predicate) but, I wonder if there's some quantitative that would be useful to include here. For example, number of lines of code added/removed/modified. I've seen some projects require additional reviewers for larger feature work, and fewer reviewers for minor bug fixes. It may be hard to craft a policy around a metric like that but, most other options I can think of turn VCS specific pretty quickly (things like commit messages that link to specific issues with a label or tag for [feat] or [bug]).